opensearch/src/main/java/org/opensearch/sql/opensearch/data/type/OpenSearchDateType.java

```diff
@@ -0,0 +1,282 @@
+@EqualsAndHashCode(callSuper = true)
+public class OpenSearchDateType extends OpenSearchDataType {
+
+  private static final OpenSearchDateType instance = new OpenSearchDateType();
+
+  private static final String FORMAT_DELIMITER = "\\|\\|";
+
+  public static final List<FormatNames> SUPPORTED_NAMED_DATETIME_FORMATS = List.of(
+      FormatNames.ISO8601,
+      FormatNames.EPOCH_MILLIS,
+      FormatNames.EPOCH_SECOND,
+      FormatNames.BASIC_DATE_TIME,
+      FormatNames.BASIC_DATE_TIME_NO_MILLIS,
+      FormatNames.BASIC_ORDINAL_DATE_TIME,
+      FormatNames.BASIC_ORDINAL_DATE_TIME_NO_MILLIS,
+      FormatNames.BASIC_WEEK_DATE_TIME,
+      FormatNames.STRICT_BASIC_WEEK_DATE_TIME,
+      FormatNames.BASIC_WEEK_DATE_TIME_NO_MILLIS,
+      FormatNames.STRICT_BASIC_WEEK_DATE_TIME_NO_MILLIS,
+      FormatNames.BASIC_WEEK_DATE,
+      FormatNames.STRICT_BASIC_WEEK_DATE,
+      FormatNames.DATE_OPTIONAL_TIME,
+      FormatNames.STRICT_DATE_OPTIONAL_TIME,
+      FormatNames.STRICT_DATE_OPTIONAL_TIME_NANOS,
+      FormatNames.DATE_TIME,
+      FormatNames.STRICT_DATE_TIME,
+      FormatNames.DATE_TIME_NO_MILLIS,
+      FormatNames.STRICT_DATE_TIME_NO_MILLIS,
+      FormatNames.DATE_HOUR_MINUTE_SECOND_FRACTION,
+      FormatNames.STRICT_DATE_HOUR_MINUTE_SECOND_FRACTION,
+      FormatNames.DATE_HOUR_MINUTE_SECOND_FRACTION,
+      FormatNames.DATE_HOUR_MINUTE_SECOND_MILLIS,
+      FormatNames.STRICT_DATE_HOUR_MINUTE_SECOND_MILLIS,
+      FormatNames.DATE_HOUR_MINUTE_SECOND,
+      FormatNames.STRICT_DATE_HOUR_MINUTE_SECOND,
+      FormatNames.DATE_HOUR_MINUTE,
+      FormatNames.STRICT_DATE_HOUR_MINUTE,
+      FormatNames.DATE_HOUR,
+      FormatNames.STRICT_DATE_HOUR,
+      FormatNames.ORDINAL_DATE_TIME,
+      FormatNames.STRICT_ORDINAL_DATE_TIME,
+      FormatNames.ORDINAL_DATE_TIME_NO_MILLIS,
+      FormatNames.STRICT_ORDINAL_DATE_TIME_NO_MILLIS,
+      FormatNames.WEEK_DATE_TIME,
+      FormatNames.STRICT_WEEK_DATE_TIME,
+      FormatNames.WEEK_DATE_TIME_NO_MILLIS,
+      FormatNames.STRICT_WEEK_DATE_TIME_NO_MILLIS
+  );
+
+  // list of named formats that only support year/month/day
+  public static final List<FormatNames> SUPPORTED_NAMED_DATE_FORMATS = List.of(
+      FormatNames.BASIC_DATE,
+      FormatNames.BASIC_ORDINAL_DATE,
+      FormatNames.DATE,
+      FormatNames.STRICT_DATE,
+      FormatNames.YEAR_MONTH_DAY,
+      FormatNames.STRICT_YEAR_MONTH_DAY,
+      FormatNames.YEAR_MONTH,
+      FormatNames.STRICT_YEAR_MONTH,
+      FormatNames.YEAR,
+      FormatNames.STRICT_YEAR,
+      FormatNames.ORDINAL_DATE,
+      FormatNames.STRICT_ORDINAL_DATE,
+      FormatNames.WEEK_DATE,
+      FormatNames.STRICT_WEEK_DATE,
+      FormatNames.WEEKYEAR_WEEK_DAY,
+      FormatNames.STRICT_WEEKYEAR_WEEK_DAY,
+      FormatNames.WEEK_YEAR,
+      FormatNames.WEEK_YEAR_WEEK,
+      FormatNames.STRICT_WEEKYEAR_WEEK,
+      FormatNames.WEEKYEAR,
+      FormatNames.STRICT_WEEKYEAR
+  );
+
+  // list of named formats that only support house/minute/second
```

Comment:
```suggestion
  // list of named formats that only support hour/minute/second
```

Reply:
Fixed in https://github.com/Bit-Quill/opensearch-project-sql/pull/268/commits/5f34b391f988e4f2e179a68bce688abc1dc15914